src/utils/dispatch.test.js

```diff
@@ -0,0 +1,125 @@
+import React, { useReducer, createContext } from 'react'
+import test from 'ava'
+import renderer from 'react-test-renderer'
+
+import {createDispatch, thunkMiddleware} from './dispatch'
+import { async, reject } from 'q'
+
+const middleware = () => {
+	return ({state, dispatch}) => {
+		return next => {
+			return action => {
+				if(action.type === 'TEST'){
+					return dispatch({
+						type: 'MIDDLEWARE',
+						payload: 'success'
+					})
+				}
+				return next(action)
+			}
+		}
+	}
+}
+
+const initState = {
+	test: 'failed',
+	switch: false
+}
+
+const reducer = (state, {type, payload}) => {
+	switch(type){
+		case 'MIDDLEWARE':
+			return {
+				...state,
+				test: payload
+			}
+		case 'DEFAULT':
```

Comment:
The case DEFAULT and default case in line 46, maybe rename DEFAULT in any other type name.